public/app/features/alerting/unified/components/rules/EditRuleGroupModal.tsx

```diff
@@ -1,18 +1,190 @@
 import { css } from '@emotion/css';
 import React, { useEffect, useMemo } from 'react';
+import { FormProvider, useForm, useFormContext } from 'react-hook-form';
 
-import { Modal, Button, Form, Field, Input, useStyles2 } from '@grafana/ui';
+import { GrafanaTheme2 } from '@grafana/data';
+import { Stack } from '@grafana/experimental';
+import { Modal, Button, Field, Input, useStyles2, Label, Icon, Tooltip, Badge } from '@grafana/ui';
+import { useAppNotification } from 'app/core/copy/appNotification';
 import { useCleanup } from 'app/core/hooks/useCleanup';
 import { useDispatch } from 'app/types';
 import { CombinedRuleGroup, CombinedRuleNamespace } from 'app/types/unified-alerting';
+import {
+  RulerRulesConfigDTO,
+  RulerRuleGroupDTO,
+  RulerRuleDTO,
+  RulerGrafanaRuleDTO,
+  RulerAlertingRuleDTO,
+} from 'app/types/unified-alerting-dto';
 
 import { useUnifiedAlertingSelector } from '../../hooks/useUnifiedAlertingSelector';
 import { updateLotexNamespaceAndGroupAction } from '../../state/actions';
 import { checkEvaluationIntervalGlobalLimit } from '../../utils/config';
 import { getRulesSourceName } from '../../utils/datasource';
 import { initialAsyncRequestState } from '../../utils/redux';
+import { parsePrometheusDuration } from '../../utils/time';
+import { DynamicTable, DynamicTableColumnProps, DynamicTableItemProps } from '../DynamicTable';
 import { EvaluationIntervalLimitExceeded } from '../InvalidIntervalWarning';
-import { evaluateEveryValidationOptions } from '../rule-editor/GrafanaEvaluationBehavior';
+import { evaluateEveryValidationOptions, MIN_TIME_RANGE_STEP_S } from '../rule-editor/GrafanaEvaluationBehavior';
+
+const MINUTE = '1m';
+interface AlertInfo {
+  alertName: string;
+  forDuration: string;
+  numberEvaluations: number;
+}
+function ForError({ message }: { message: string }) {
+  return <Badge color="orange" icon="exclamation-triangle" text={'Error'} tooltip={message} />;
+}
+
+const getNumberEvaluationsToStartAlerting = (forDuration: string, currentEvaluation: string) => {
+  const evalNumber = safeParseDurationstr(currentEvaluation);
```

Comment:
nit: maybe rename to `evalMs` or `evalNumberMs` to indicate this is number of milliseconds?